python/extensions/pip.bzl

```
@@ -316,9 +437,55 @@ Each `pip.parse()` call configures a particular Python version. Multiple calls
 can be made to configure different Python versions, and will be grouped by
 the `hub_name` argument. This allows the same logical name, e.g. `@pip//numpy`
 to automatically resolve to different, Python version-specific, libraries.
+
+pip.whl_mods:
+This tag class is used to create different wheel modification JSON files.  These files
+contain directives that are used by the wheel_installer.py during the creation of 
+wheels.
 """,
     implementation = _pip_impl,
     tag_classes = {
-        "parse": tag_class(attrs = _pip_parse_ext_attrs()),
+        "parse": tag_class(
+            attrs = _pip_parse_ext_attrs(),
+            doc = """\
+This tag class is used to create a pip hub and all of the spokes that are part of that hub.
+This tag class reuses the pip attributes that are found in 
+@rules_python//python/pip_install:pip_repository.bzl
```

The hub and spoke part isn't very relevant to users (they can't control it, nor do they really see it. The available attributes are self-documenting in the `attrs` structure, plus we don't support all the same attributes (e.g incompatible_generate_aliases), so pointing users to that is a bit misleading.

```suggestion
This tag class resolves requirements.txt into dependencies usable by Bazel.
```

---

I think referring to the pip repo attrs is ok, but saying/implying it reuses them or supports all of them isn't (it just isn't accurate/true). i.e. "most of the pip attrs are supported" but not "all the pip attrs are supported"

---

I understand.  I will figure out how to document it better.